File: lab3/lab3 app/app/src/main/java/com/pjatk/bmi/ui/fragment/ChartFragment.java
package com.pjatk.bmi.ui.fragment;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.data.LineData;
import com.github.mikephil.charting.data.LineDataSet;
import com.pjatk.bmi.R;

import java.util.ArrayList;

public class ChartFragment extends Fragment {

    LineChart lineChart;
    LineData lineData;
    LineDataSet lineDataSet;
    ArrayList lineEntries;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_chart, container, false);

        lineChart = view.findViewById(R.id.lineChart);
        lineChart.getXAxis().setPosition(XAxis.XAxisPosition.BOTTOM);
        lineChart.getAxisLeft().setDrawGridLines(false);
        lineChart.getAxisRight().setDrawGridLines(false);
        lineChart.getDescription().setText("BMI / Months");

        getEntries();

        lineDataSet = new LineDataSet(lineEntries, "changes in BMI over the year");
        lineDataSet.setColor(Color.RED);
        lineDataSet.setLineWidth(1f);
        lineDataSet.setValueTextColor(Color.BLACK);

        lineData = new LineData(lineDataSet);
        lineChart.setData(lineData);
        lineChart.invalidate();
        lineDataSet.setValueTextColor(Color.BLACK);
        lineDataSet.setValueTextSize(18f);



        return view;
    }

    private void getEntries() {
        lineEntries = new ArrayList();
        lineEntries.add(new Entry(1, 20.5f));
        lineEntries.add(new Entry(2, 19.9f));
        lineEntries.add(new Entry(3, 18.0f));
        lineEntries.add(new Entry(4, 18.5f));
        lineEntries.add(new Entry(5, 19.0f));
        lineEntries.add(new Entry(6, 19.2f));
        lineEntries.add(new Entry(7, 19.5f));
        lineEntries.add(new Entry(8, 19.5f));
        lineEntries.add(new Entry(9, 19.3f));
        lineEntries.add(new Entry(10, 19.0f));
        lineEntries.add(new Entry(11, 18.7f));
        lineEntries.add(new Entry(12, 18.2f));
    }



}

File: lab4/lab4 app/app/src/main/java/com/pjatk/bmi/ui/fragment/ChartFragment.kt
package com.pjatk.bmi.ui.fragment

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.github.mikephil.charting.charts.LineChart
import com.github.mikephil.charting.components.XAxis
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.data.LineData
import com.github.mikephil.charting.data.LineDataSet
import com.pjatk.bmi.R

class ChartFragment : Fragment() {
    private lateinit var lineChart: LineChart
    private lateinit var lineData: LineData
    private lateinit var lineDataSet: LineDataSet
    private lateinit var lineEntries: ArrayList<Entry>

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_chart, container, false)

        lineChart = view.findViewById(R.id.lineChart)
        lineChart.xAxis.position = XAxis.XAxisPosition.BOTTOM
        lineChart.axisLeft.setDrawGridLines(false)
        lineChart.axisRight.setDrawGridLines(false)
        lineChart.description.text = "BMI / Months"

        getEntries()

        lineDataSet = LineDataSet(lineEntries, "changes in BMI over the year")
        lineDataSet.color = Color.RED
        lineDataSet.lineWidth = 1f
        lineDataSet.valueTextColor = Color.BLACK

        lineData = LineData(lineDataSet)
        lineChart.data = lineData
        lineChart.invalidate()
        lineDataSet.valueTextColor = Color.BLACK
        lineDataSet.valueTextSize = 18f

        return view
    }

    private fun getEntries() {
        lineEntries = arrayListOf(
                Entry(1f, 20.5f),
                Entry(2f, 19.9f),
                Entry(3f, 18.0f),
                Entry(4f, 18.5f),
                Entry(5f, 19.0f),
                Entry(6f, 19.2f),
                Entry(7f, 19.5f),
                Entry(8f, 19.5f),
                Entry(9f, 19.3f),
                Entry(10f, 19.0f),
                Entry(11f, 18.7f),
                Entry(12f, 18.2f)
        )
    }
}
